haskell-library/README.md

```diff
@@ -0,0 +1,22 @@
+Template for building and testing haskell library with multiple versions of GHC via haskell.nix for Buildkite, Gitlab or GitHub CI
+
+## How to use this template
+
+- The recommended way to use this template (if you have nix flakes enabled) is `nix flake init -t github:serokell/templates#haskell-library`
+
+- Otherwise:
+    1. Copy `flake.nix` file to your repository, it contains all nix dependencies and definitions.
+
+    2. Copy `default.nix` and `shell.nix` files which provide compatibility with non-flake nix interface.
+
+    3. Get unstable nix with `nix-command flakes` experimental features enabled, and run `nix flake update` to generate `flake.lock` file. This file pins the latest versions of the nix dependencies.
+
+    4. Copy pipeline configuration: `.gitlab-ci.yml` for Gitlab, `.buildkite/pipeline.yml` for Buildkite or `.github/workflows/check.yml` for GitHub.
+
+- Afterwards, adjust the template for your project:
+
+    1. Replace `pataq-package` in `flake.nix` with your haskell library name (usually specified in `package.yaml`). And replace `ghc-versions` with the list of GHC versions that will be used to build and test your library.
```

Comment:
Another common use-case when a single repo provides multiple packages, e.g. https://github.com/haskell-servant/servant. I think it's worth mentioning what needs to be updated in order to have a CI setup for multiple packages at once